### DISCUSSION TITLE
Changes to Juggle in Code

### DISCUSSION THREAD

**Discussion by Kokorz** [ikemen-engine/Ikemen-GO#878] [Feature Request] [answered]
I think it's kinda silly that only juggle isn't changeable seperately via an sctrl or trackable via a trigger while nearly, if not every parameter in the Statedef SCTRL is manipulatable otherwise.

Thus, I have 4 suggestions.

**juggle (trigger)**

Returns how many juggle points the character (or helper, or partner, or enemy) has left.

**totaljuggle (trigger)**

Returns the total amount of juggle points the character (or opponent, yadda yadda) has.

**JuggleAdd (SCTRL)**

Allows you to add or subtract juggle points whenever a certain condition is true.

Required Parameters:
> value = <_int_>

**JuggleSet (SCTRL)**

Sets the juggle points to a specified value.

Required Parameters:
> value = <_int_>

Optional Parameters:

> applytototal = <_boolean_>

 (Changes the total juggle points rather than it change one instance (or in other words, changes the air.juggle value rather than your juggle points value.)

Maybe a parameter that subtracts/adds juggle in the hitdef would be neat too. (You'd think air.juggle would do this but nope)

In terms of use cases, rather than switch to another state to subtract juggle again, you can use a JuggleAdd with animelem as the trigger to subtract at that frame.

You could also use the juggle trigger with a juggleset sctrl and a variable/map for in the case that if you set your attack's attribute to a throw and do a grab in the middle of a combo, it won't reset your juggle back to the max value but rather you can set it back to what the variable got (the previous juggle value before the grab) and subtract more.

(This can all be rendered useless by just making your own juggle system but it probably wouldn't hurt to make the existing system more bearable)

**Comment by Windblade-GR01** [comment, marked as answer]
I did just made juggle trackable via the ` jugglePoints` trigger a week ago.
This sound like a good idea to expand it even more.
